.github/workflows/main.yml

```diff
@@ -0,0 +1,45 @@
+name: Continuous Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  documentation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone @api3/promise-utils
+        uses: actions/checkout@master
+      - name: Check hyperlinks
+        uses: gaurav-nelson/github-action-markdown-link-check@v1
+
+  lint-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
+    name: Build (${{ matrix.node-version }})
+    steps:
+      - name: Clone @api3/promise-utils
+        uses: actions/checkout@v2
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install deps and build (with cache)
+        uses: bahmutov/npm-install@v1
```

Comment:
Does this action respect the node version set above? I can't find anything around this and all of their examples don't have the `setup-node` step above, so it must be doing something to make sure node is available? e.g. https://github.com/bahmutov/npm-install/blob/master/.github/workflows/example-yarn.yml

I assume it does respect a previously set node version?

Reply:
I was trying to debug why the tests fail on CI, but then I figured it out. Consider this irrelevant.